# Descriptive Statistics

#### Week 2 - Day 1

>Note: I chose a different dataset than last week, with the aim of practicing new datasets and reinforcing last week concepts at the same time.

**Dataset**: Medical Cost Personal Datasets (Insurance).

In [2]:
import pandas as pd
import numpy as np

df = pd.read_csv("../data/insurance.csv")
df.head()

,age,sex,bmi,children,smoker,region,charges
0,19,female,27.900,0,yes,southwest,16884.92400
1,18,male,33.770,1,no,southeast,1725.55230
2,28,male,33.000,3,no,southeast,4449.46200
3,33,male,22.705,0,no,northwest,21984.47061
4,32,male,28.880,0,no,northwest,3866.85520


## Hands-On Lab:

#### Step 1 & 2: 

> Load "charges" column and calculate mean, median, mode, std and IQR.

In [5]:
charges = df["charges"]

mean = charges.mean()
median = charges.median()
mode = charges.mode()
standard_deviation = charges.std()

Q1 = charges.quantile(0.25)
Q3 = charges.quantile(0.75)
IQR = Q3 - Q1

print("Mean =", mean)
print("Median =", median)
print("Standard Deviation =", standard_deviation)
print("IQR =", IQR)

Mean = 13270.422265141257
Median = 9382.033
Standard Deviation = 12110.011236694001
IQR = 11899.625365


#### Step 3:

> Which better represents a "typical" value, mean or median?

In [6]:
print("Mean =", mean)
print("Median =", median)
print("The Difference =", mean - median)

Mean = 13270.422265141257
Median = 9382.033
The Difference = 3888.389265141257


#### We notice the mean (13270) is higher than median (9382), So the **median** is better to represents a 'typical' value.

#### Step 4:

> Summary the column's center and spread:

- In general, patients pay around 13270 for medical charges, but the median is only around 9382, meaning half of all patients pay less than that. 
This tells us a small group of patients with very high bills is pulling the average up. Because of this, the median gives a better picture of what a "typical" patient actually pays.
The data is also spread out a lot (the standard deviation is close to $12110), which confirms that charges vary greatly from one to another.